registration_rust/src/main.rs

```diff
@@ -0,0 +1,612 @@
+use std::{env, fmt};
+use std::error::Error;
+use std::fmt::{Debug, Display, Formatter};
+use axum::{Json, Router, routing::get};
+use axum::extract::{FromRef, State};
+use axum::http::{header, StatusCode};
+use axum::response::IntoResponse;
+use axum::routing::{post, put};
+use rand::distributions::Alphanumeric;
+use rand::Rng;
+use redis::{Client, RedisError, AsyncCommands, RedisResult};
+use serde::{Deserialize, Serialize};
+use tokio::net::TcpListener;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+
+enum PestoError {
+    UserNotFound,
+    EmailAlreadyExists,
+    ParseError,
+    RedisError(RedisError),
+}
+
+impl Display for PestoError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            PestoError::UserNotFound => write!(f, "User not found"),
+            PestoError::EmailAlreadyExists => write!(f, "Email already exists"),
+            PestoError::RedisError(e) => write!(f, "Redis error: {}", e),
+            PestoError::ParseError => write!(f, "Parse error"),
+        }
+    }
+}
+
+
+#[derive(Clone)]
+struct ApplicationState<Command: AsyncCommands + Clone> {
+    waiting_list_service: WaitingListService<Command>,
+    approval_service: ApprovalService<Command>,
+    trial_service: TrialService<Command>,
+    mailersend_client: MailersendClient,
+}
+
+impl<Command: AsyncCommands + Clone> Debug for ApplicationState<Command> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "ApplicationState")
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+struct HumanUser {
+    name: String,
+    email: String,
+    building: Option<String>,
+    calls: i64,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+struct TokenValue {
+    pub token: Option<String>,
+    pub user_email: String,
+    pub monthly_limit: i64,
+    pub revoked: bool,
+}
```

Comment:
Might need to change this reflecting #172

Reply:
Also, need to find a way to not render out the `token`. Probably with https://serde.rs/field-attrs.html#skip